app/models/location.rb

```diff
@@ -32,12 +43,16 @@ def attributes
       field :persistent
       field :traits
       field :tags
-      field :icon
+      field :icons
     end
   end
 
   private
 
+  def icon_url
+    self.icons.first.icon_image.url unless self.icons.empty?
```

Comment:
if Location has_many icons, then you're only fetching the first one. It seems that you would want this to be an array of all of the urls 
```
def icon_urls
  icons.map { |i| i.icon_image.url } unless icons.empty?
end
```